internal/config/db.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"log"
+
+	"github.com/b-open/jobbuzz/pkg/model"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
+
+var Db gorm.DB
+
+// initialise DB once
+func InitDb() {
+	// TODO: 12-Factor the connection string
+	dsn := "jobbuzz:secret@tcp(127.0.0.1:3306)/jobbuzz?charset=utf8mb4&parseTime=True&loc=Local"
+	db, err := gorm.Open(mysql.Open(dsn), &gorm.Config{})
+
+	if err != nil {
+		log.Fatal("Fail to connect to DB", err)
+	}
+
+	Db = *db
+}
+
+// reusing the same DB instance
+func GetDb() *gorm.DB {
+	return &Db
+}
+
+func MigrateDb(db *gorm.DB) {
+	err := db.AutoMigrate(&model.Job{})
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Return the error and have the code calling this do the log.fatal

If this is called further down a chain of functions, you wouldn't be able to trace it down from the logs.

Or if the upper level function could have attached more info that this function couldn't have.

Reply:
You haven't changed it to return the error?